### PR TITLE
Remove internal access modifier from ComponentFoundation

### PR DESCRIPTION
**Pull request by HannahKiekens**

### CODE REVIEW

**Comment by HannahKiekens** [comment]
Context:
I am trying to write a "mdc-menu-surface" like component in the same coding style as Material.Blazor
These internal access modifiers prevent me from subclassing ComponentFoundation and using it in a similar fashion as MBMenu

File: Material.Blazor/Model/MBEnumerations.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@ namespace Material.Blazor
     /// <code>@attributes="<see cref="AttributesToSplat"/>(<see cref="IdClassAndStyleOnly"/> | <see cref="HtmlExcludingIdClassAndStyle"/>)"</code>
     /// </summary>
     [Flags]
-    internal enum SplatType : ushort
+    public enum SplatType : ushort
```
Comment (stefanloerwald) [inline Material.Blazor/Model/MBEnumerations.cs:10]: This really is an internal enum, but I see why you need it. Would it maybe make sense to make it a nested type in ComponentFoundation and would it be possible to have it `internal protected`?

**Comment by HannahKiekens** [comment]
I'll put up a new approach for my problem

**Comment by stefanloerwald** [comment]
FYI: I'm working on some library-wide changes that get rid of the `SplatType` enum entirely.

**Pull request closed without merging**